Bluetooth: controller: df: Fix check for saturated IQ samples.

jakkra opened the pull request:
When an int16_t I or Q value is input to IQ_CONVERT_12_TO_8_BIT the compiler would
not make the correct comparison with IQ_SAMPLE_STATURATED_16_BIT causing saturated
IQ samples never being found.

This macro is not doing what is expected: https://github.com/zephyrproject-rtos/zephyr/blob/a8ffd192813a5e8ac1f5bf89123d3009f5d6539e/subsys/bluetooth/controller/ll_sw/nordic/lll/lll_df_types.h#L82

In the end it boils down to:
```
(-32768 == 0x8000) => false
(-32768 == (int16_t)0x8000) => true (this is what we want)
``` 
Hence this fix.

cvinayak commented:
@jakkra Fix the below compliance error in the signed-off-by:
```
c7cc7c6ffd75f738e9ecc5b8664a0cd782408362: author email (jakob.krantz <jakob.krantz@u-blox.com>) does not follow the syntax: First Last <email>.
```

jakkra commented:
Fixed

cvinayak commented:
@jakkra try the below
```
git config --global user.name "Jacob Krantz"
git config --global user.email "jakob.krantz@u-blox.com"
git commit --amend --reset-author -s
```

ppryga commented on subsys/bluetooth/controller/ll_sw/nordic/lll/lll_df_types.h:
```diff
@@ -75,7 +75,7 @@ struct lll_df_adv_cfg {
  */
-#define IQ_SAMPLE_STATURATED_16_BIT 0x8000
+#define IQ_SAMPLE_STATURATED_16_BIT ((int16_t)0x8000)
 #define IQ_SAMPLE_STATURATED_8_BIT 0x80
```
Could you add `((int8_6)0x80)` to this line? That will make sure there is no similar issue if one uses the second macro in a context where integer promotions are applied.

jakkra replied:
Actually did that first, but removed, have now added it again and hopefully fixed the signed off message

jakkra committed:
Bluetooth: controller: df: Fix check for saturated IQ samples.

When an int16_t I or Q value was input to IQ_CONVERT_12_TO_8_BIT
the compiler would not make the correct comparison with
IQ_SAMPLE_STATURATED_16_BIT causing saturated IQ samples never
being found.

Signed-off-by: Jakob Krantz <jakob.krantz@u-blox.com>
